production/ksonnet/loki/bloom-gateway.libsonnet

```
@@ -0,0 +1,130 @@
+{
+  local k = import 'ksonnet-util/kausal.libsonnet',
+  local container = k.core.v1.container,
+  local containerPort = k.core.v1.containerPort,
+  local pvc = k.core.v1.persistentVolumeClaim,
+  local service = k.core.v1.service,
+  local statefulSet = k.apps.v1.statefulSet,
+  local volume = k.core.v1.volume,
+  local volumeMount = k.core.v1.volumeMount,
+
+  _config+:: {
+    bloom_gateway+: if !$._config.use_bloom_filters then {} else {
+      // TODO(salvacorts): Configure autoscaling
+      replicas: error 'missing replicas',
+
+      ////** Storage **////
+      // if true, the host needs to have local SSD disks mounted, otherwise PVCs are used
+      use_local_ssd: false,
+      // PVC config
+      pvc_size: if !self.use_local_ssd then error 'bloom_gateway.pvc_size needs to be defined when using PVC' else '',
+      pvc_class: if !self.use_local_ssd then error 'bloom_gateway.pvc_class needs to be defined when using PVC' else '',
+      // local SSD config
+      hostpaths: if self.use_local_ssd then error 'bloom_gateway.hostpaths needs to be defined when using local SSDs' else [],
+      node_selector: if self.use_local_ssd then error 'bloom_gateway.node_selector needs to be defined when using local SSDs' else {},
+      tolerations: if self.use_local_ssd then error 'bloom_gateway.tolerations needs to be defined when using local SSDs' else [],
+    },
+  },
+
+  local name = 'bloom-gateway',
+
+  local paths = std.range(0, std.length($._config.bloom_gateway.hostpaths) - 1),
+
+  local volumeNames = [
+    '%s-data-%d' % [name, x]
+    for x in paths
+  ],
+
+  local volumes =
+    if $._config.bloom_gateway.use_local_ssd
+    then [
+      volume.fromHostPath(volumeNames[x], $._config.bloom_gateway.hostpaths[x])
+      for x in paths
+    ]
+    else [],
+
+  local volumeMounts = [
+    volumeMount.new(volumeNames[x], '/data%d' % [x])
+    for x in paths
+  ],
+
+  bloom_gateway_args:: $._config.commonArgs {
+    target: 'bloom-gateway',
+  },
+
+  bloom_gateway_ports:: [
+    containerPort.new(name='http-metrics', port=$._config.http_listen_port),
+    containerPort.new(name='grpc', port=9095),
+  ],
+
+  bloom_gateway_data_pvc:: if !$._config.use_bloom_filters || !$._config.bloom_gateway.use_local_ssd then null else
```

bug: PVC is null if you *arent* using local SSDs

```suggestion
  bloom_gateway_data_pvc:: if !$._config.use_bloom_filters || $._config.bloom_gateway.use_local_ssd then null else
```